iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/topology/CreateFeederBay.java

```diff
@@ -0,0 +1,183 @@
+ */
+package com.powsybl.iidm.modification.topology;
+
+import com.powsybl.commons.PowsyblException;
+import com.powsybl.commons.reporter.Reporter;
+import com.powsybl.computation.ComputationManager;
+import com.powsybl.iidm.modification.NetworkModification;
+import com.powsybl.iidm.network.*;
+import com.powsybl.iidm.network.extensions.BusbarSectionPosition;
+import com.powsybl.iidm.network.extensions.ConnectablePosition;
+import com.powsybl.iidm.network.extensions.ConnectablePositionAdder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static com.powsybl.iidm.modification.ModificationReports.*;
+import static com.powsybl.iidm.modification.topology.TopologyModificationUtils.createNBDisconnector;
+import static com.powsybl.iidm.modification.topology.TopologyModificationUtils.createNodeBreakerSwitches;
+
+/**
+ * This method adds a new injection bay on an existing voltage level. The voltage level should be described
+ * in node/breaker topology.
+ * @author Coline Piloquet <coline.piloquet at rte-france.com>
+ */
+public class CreateFeederBay implements NetworkModification {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CreateFeederBay.class);
+
+    private final InjectionAdder<?> injectionAdder;
+    private final String bbsId;
+    private final int injectionPositionOrder;
+    private final ConnectablePosition.Direction injectionDirection;
+
+    /**
+     * Constructor.
+     *
+     * @param injectionAdder         The injection adder.
+     * @param bbsId                  The ID of the existing bus bar section of the voltage level voltageLevelId where we want to connect the injection.
+     *                               Please note that there will be switches between this bus bar section and the connection point of the injection. This switch will be closed.
+     * @param injectionPositionOrder The order of the injection to be attached from its extension {@link ConnectablePosition}.
+     * @param injectionDirection     The direction of the injection to be attached from its extension {@link ConnectablePosition}.
+     */
+    public CreateFeederBay(InjectionAdder<?> injectionAdder, String bbsId, int injectionPositionOrder, ConnectablePosition.Direction injectionDirection) {
+        this.injectionAdder = Objects.requireNonNull(injectionAdder);
+        this.bbsId = Objects.requireNonNull(bbsId);
+        this.injectionPositionOrder = injectionPositionOrder;
+        this.injectionDirection = Objects.requireNonNull(injectionDirection);
+    }
+
+    public CreateFeederBay(InjectionAdder<?> injectionAdder, String bbsId, int injectionPositionOrder) {
+        this(injectionAdder, bbsId, injectionPositionOrder, ConnectablePosition.Direction.BOTTOM);
+    }
+
+    @Override
+    public void apply(Network network, ComputationManager computationManager) {
```

Comment:
You have to move these 2 apply methods in the NetworkModification interface otherwise the added throwException is a breaking change

Reply:
@colinepiloquet can you take this comment into account?

Reply:
What do you mean? Aren't they already in the interface?

Reply:
@geofjamg For all implementations, we have this override:
````
    @Override
    public void apply(Network network, ComputationManager computationManager) {
        apply(network);
    }
````
So if we want to put this default behavior in the interface, it should be done in another PR.
The other override is needed because this hypothesis does not implement the apply(network, computationManager).
````
    @Override
    public void apply(Network network) {
        apply(network, false, Reporter.NO_OP);
    }
````